Tests/OpenAPIRuntimeTests/EventStreams/Test_ServerSentEventsDecoding.swift

```diff
@@ -16,17 +16,19 @@ import XCTest
 import Foundation
 
 final class Test_ServerSentEventsDecoding: Test_Runtime {
-    func _test(input: String, output: [ServerSentEvent], file: StaticString = #filePath, line: UInt = #line)
+    func _test(input: String, output: [ServerSentEvent], file: StaticString = #filePath, line: UInt = #line, while predicate: (ArraySlice<UInt8>) -> Bool = { _ in true }, eventCountOffset: Int = 0)
```

Comment:
I had initially tried it without `eventCountOffset `, but the numbers didn’t match for me. 

```swift
    try await _test(
            input: #"""
                data: hello
                data: world

                data: [DONE]

                data: hello2
                data: world2


                """#,
            output: [
                .init(data: "hello\nworld")
            ],
            while: { incomingData in
                incomingData != ArraySlice<UInt8>(Data("[DONE]".utf8))
            },
            eventCountOffset: -2
```

For instance, without `eventsOffset`, this test case will return 3 instead of 1.

Reply:
I suspect that's because the predicate isn't actually passed into the tested function `asDecodedServerSentEvents` below

Reply:
Oh, that one flew right past me. Working now!